src/main/java/io/odpf/firehose/sink/common/blobstorage/s3/S3.java

```diff
@@ -0,0 +1,87 @@
+package io.odpf.firehose.sink.common.blobstorage.s3;
+
+
+import io.odpf.firehose.config.S3Config;
+import io.odpf.firehose.sink.common.blobstorage.BlobStorage;
+import io.odpf.firehose.sink.common.blobstorage.BlobStorageException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.core.client.config.ClientOverrideConfiguration;
+import software.amazon.awssdk.core.exception.SdkClientException;
+import software.amazon.awssdk.core.exception.SdkServiceException;
+import software.amazon.awssdk.core.retry.RetryPolicy;
+import software.amazon.awssdk.core.sync.RequestBody;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.HeadBucketRequest;
+import software.amazon.awssdk.services.s3.model.NoSuchBucketException;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+import software.amazon.awssdk.services.s3.model.S3Exception;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+public class S3 implements BlobStorage {
+    private static final Logger LOGGER = LoggerFactory.getLogger(S3.class);
+    private final S3Client s3Client;
+    private final String bucketName;
+
+    public S3(S3Config s3Config) {
+        RetryPolicy retryPolicy = RetryPolicy.builder()
```

Comment:
@akhildv can we also add retry policy with exponential back off? We can add similar config as GCS.

Reply:
@lavkesh @ravisuhag added more retry policies , backoff stratergy and updates docs for the same. please review